02-peer-protocol.md

```diff
@@ -348,8 +348,8 @@ The sender:
 avoid double-spending of the funding transaction.
   - MUST set `channel_reserve_satoshis` greater than or equal to `dust_limit_satoshis` from the `open_channel` message.
   - MUST set `dust_limit_satoshis` less than or equal to `channel_reserve_satoshis` from the `open_channel` message.
-  - if it sets `channel_type`:
-    - MUST set it to the `channel_type` from `open_channel`
+  - if `option_channel_type` was negotiated:
+    - MUST set `channel_type` to the `channel_type` from `open_channel`
```

Comment:
Hmmm, this isn't quite right, anymore. Recall that the feature was added after the `channel_type` stuff was, thus, the "authoritative" data is the `channel_type` field, not the feature. It feels strange to remove the mention of setting the type here if you don't negotiate the feature.

Reply:
> Recall that the feature was added after the channel_type stuff was, thus, the "authoritative" data is the channel_type field, not the feature.

That's true, but since now all implementations do use the feature bit, this is a historical artifact that can be removed from the spec, isn't it? Now that there is a feature bit, it does simplify requirements to always use it

Reply:
Not really? There are releases of node software that does not set the feature bit and expects the `channel_type` flags as sent to determine the channel type. I'd rather keep the `channel_type` so that we're clear on how things work, including with some nodes on the network.

Reply:
Ok, that sounds reasonable (and matches my first version of this PR). @niftynei does that work for you, since you requested changes in the initial version?

Reply:
it feels like the mention of the feature flag is unnecessary? is there ever a case where the feature flag will be set but there won't be a `channel_type` to mirror back in the accept?

maybe i'm unclear on what case we're solving for here, apologies.

Reply:
That's the whole point of this PR, previously we didn't *require* the `channel_type` be set even if the feature flag is, but, as you note its somewhat nonsensical for that to happen. This PR simply fixes that oversight and says "yea, if you negotiate the feature that says you support this, you MUST support this, yo"

Reply:
Then the change belongs on the open channel message, not the accept message.

The accept message behavior is unchanged and the existing spec should be sufficient.

Reply:
No? The change here is about the `channel_type` field in the accept message. The current spec only says that if you set `channel_type` in accept then it has to be a specific value, it never says anything about when you have to include the `channel_type` to begin with.